jedi/inference/references.py

```diff
@@ -113,25 +114,28 @@ def _find_global_variables(names, search_name):
                     yield n
 
 
-def find_references(module_context, tree_name):
+def find_references(module_context, tree_name, all_scopes=True):
     inf = module_context.inference_state
     search_name = tree_name.value
 
     # We disable flow analysis, because if we have ifs that are only true in
     # certain cases, we want both sides.
     try:
         inf.flow_analysis_enabled = False
-        found_names = _find_defining_names(module_context, tree_name)
+        found_names = _find_defining_names(module_context, tree_name, all_scopes=all_scopes)
```

Comment:
I disagree with this. it should always find the defining names in all files. We just shouldn't return them. We would probably need to filter them in the end.

Reply:
From an LSP perspective, this function would be used for [documentHighlight](https://microsoft.github.io/language-server-protocol/specification#textDocument_documentHighlight), which is generally called on `CursorHold` in Vim:

```vim
autocmd CursorHold * silent call CocActionAsync('highlight')
```

Ideally, module-only references wouldn't need to do any unnecessary searching, which would almost certainly contribute to unreasonably high CPU usage with the documentHighlight use-case.

Reply:
This is not an additional search. This is the search for the actual definition, which is important. Otherwise you won't be able to e.g. find all references of str in your file.

Reply:
Ahh, that makes sense.